## Minipreguntas
Intenta razonar el resultado antes de ejecutarlo:

**¿Cuál es el output del siguiente código?**

In [3]:
print("Hola mundo"*3)

Hola mundoHola mundoHola mundo


In [4]:
a = [1, 2, 3]
b = a.append(4)
print(a, b)

[1, 2, 3, 4] None


In [6]:
a.append(4)

In [5]:
# 
a = [1, 2, 3, 4]
a.insert(1, a.pop(2))
print(a)

[1, 3, 2, 4]


In [8]:
x = 5
y = 10
x, y = y, x
print(x, y)

10 5


In [9]:
def funcion(num):
    return num * 3 if num < 10 else num * 2

print(funcion(funcion(4)), funcion(funcion(10)))

24 40


In [10]:
lista = [1, 2, 3, 4, 5]
lista[:3]

[1, 2, 3]

In [11]:
lista = [1, 2, 3, 4, 5]
lista[:3] = lista[-1:-4:-1]
print(lista)

[5, 4, 3, 4, 5]


In [ ]:
numero = 5

def funcion():
    numero = 3

funcion()   

print(numero)

In [12]:
lista_1 = [1, 2, 3]
lista_2 = lista_1

lista_2[0] = 4

print(lista_1)

[4, 2, 3]


## Ejercicios

**1.-** Multiplica 2 números y muestra su resultado por pantalla

In [13]:
print(4*5)

20


**2.-** Escribe una función que te pida 2 números y te devuelva su suma por pantalla

In [14]:
def suma(num1, num2):
    print(num1+num2)

suma(45,5)

50


**3.-** Crea una variable string con los números del 1 al 20 y muestra solo los índices pares

In [19]:
string_1_20 = "1234567891011121314151617181920"
string_1_20[1::2]

'246811111111112'

**3.-b** Crea la variable anterior pero hasta el 100 de manera automatizada. Haz que los números estén separados por un espacio.

In [23]:
string_1_100 = ""

for numero in range(1,100+1):
    string_1_100 += str(numero)
    string_1_100 += " "

string_1_100

'1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40 41 42 43 44 45 46 47 48 49 50 51 52 53 54 55 56 57 58 59 60 61 62 63 64 65 66 67 68 69 70 71 72 73 74 75 76 77 78 79 80 81 82 83 84 85 86 87 88 89 90 91 92 93 94 95 96 97 98 99 100 '

In [26]:
" ".join([str(num) for num in range(1,101)])

'1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40 41 42 43 44 45 46 47 48 49 50 51 52 53 54 55 56 57 58 59 60 61 62 63 64 65 66 67 68 69 70 71 72 73 74 75 76 77 78 79 80 81 82 83 84 85 86 87 88 89 90 91 92 93 94 95 96 97 98 99 100'

**4.-** Dada una lista divide todos sus elementos entre 5 y muestralo por pantalla

In [27]:
a = [10,15,20,25,30,35,40,45,50]

In [28]:
for num in a:
    print(num/5)

2.0
3.0
4.0
5.0
6.0
7.0
8.0
9.0
10.0


**5.-** Crea una función que compruebe si el primer y el último elemento de una lista son iguales.

In [29]:
def chequeo(lista):
    return lista[0] == lista[-1]

chequeo(a) 

False

**6.-** Devuelve cuantas veces aparece la palabra "por" en el string

In [30]:
x_string = '''Por una mirada, un mundo,
            por una sonrisa, un cielo,
            por un beso… ¡yo no sé
            qué te diera por un beso!'''

In [36]:
x_string.lower().count('por')

4

**7.-** Crea una función que dado un número entero devuelva este patrón
```
1
22
333
4444
55555
```

In [39]:
def repetir_cifras(numero):
    return print(str(numero)*numero)

repetir_cifras(5)

55555


**8.-** Escribe una función que compruebe si un número es capicúa

In [ ]:
x = 555

In [ ]:
def capicua(numero):
    for i in range(len(str(numero))//2):
        if str(numero)[i] != str(numero)[-i]:
            return False
    
    return True

In [ ]:
def capicua(numero):
    return str(numero) == str(numero)[::-1]

**9.-** Escribe una lista a partir de otras 2 lista cumpliendo las siguientes condiciones:
- Cogemos los números pares de la lista 1
- Cogemos los números impares de la lista 2

```
lista_1 = [20,25,30,45,80]
lista_2 = [2,4,5,6,7]
>>>>>>>>>
lista_resultado = [20,30,80,5,7]
```

In [42]:
lista_1 = [20,25,30,45,80]
lista_2 = [2,4,5,6,7]

lista_vacia = []

for i in lista_1:
    if i%2 == 0:
        lista_vacia.append(i)

for i in lista_2:
    if i%2 != 0:
        lista_vacia.append(i)

lista_vacia


[20, 30, 80, 5, 7]

In [44]:
[x for x in lista_1 if x%2==0] + [x for x in lista_2 if x%2!=0]

[20, 30, 80, 5, 7]

**10.-** Crea una función que calcule el sueldo  después de impuestos y que porcentaje se paga del mismo, basate en las siguientes reglas:
- Primer tramo hasta 12450 euros en el que se paga un 19%.
- Segundo tramo hasta 20200 euros con tipo del 24%
- Tercer tramo hasta 35200 euros con un 30%.
- Cuarto tramo hasta 60000 euros con un 37%
- Quinto tramo hasta 300000 euros con un 45%
- Sexto tramo a partir de 300000 euros de 47%.

Ejemplo de cálculo: persona que cobra 30000€:
-  12450*(0.19)+(20200-12450)* (0.24)+(30000-20200) *(0.30)
- Total a pagar : 7165.5 (~24%)

**10-b** Usa la función anterior

**11.-** Genera un código al que le podamos pasar un string de cualquier longitud, separe el string en palabras y luego nos imprima por pantalla aquellas ‘palabras’ que en realidad sean números. Utiliza el siguiente string:

***“01 In the beginning the Universe was created. 02 This had made many people very angry 03 and has been widely regarded as a bad move.”***

**12.-** A veces necesitamos tener vectores o listas que sean de una longitud concreta. Crea una **función** a la que le pases una lista (vacía o con elementos), un número que represente la longitud deseada y un elemento que sea el valor con el que ‘rellenar’ hasta la longitud deseada.

**12.-b** Modifica la función anterior para poder elegir si introducimos el relleno antes o después de la lista proporcionada:

**13.-** Crea una función que dado un número te devuelva una lista con todos los números enteros positivos impares hasta dicho número pero que:
- No incluya los números divisibles por 7.
- No incluya los números capicúa pero sí los números en los que todas las cifras sean iguales (ej. 33 pero no 121)
